### PR TITLE
deps: update to Go 1.20

### PR DESCRIPTION
**Pull request by katexochen**
Signed-off-by: Paul Meyer <49727155+katexochen@users.noreply.github.com>

### Proposed change(s)

- Changelog: https://tip.golang.org/doc/go1.20
- Replaced uber's multierr package with new native errors.Join
- Replaced deprecated math/rand.Seed
- Use -C flag for Go commands in CI

### Checklist

- [x] Add labels (e.g., for changelog category)

### CODE REVIEW

File: internal/config/config_test.go [hunk 1]
```diff
@@ -185,28 +184,21 @@ func TestNewWithDefaultOptions(t *testing.T) {
 }
 
 func TestValidate(t *testing.T) {
-	const defaultErrCount = 21 // expect this number of error messages by default because user-specific values are not set and multiple providers are defined by default
-	const azErrCount = 9
-	const gcpErrCount = 6
-
```
Comment (datosh) [inline internal/config/config_test.go:191]: I don't think the tests will actually catch changes / errors now. The problem we worked around was that the `Default()` config itself was not a valid config. Any tests relativ to the config generated by `Default()` would also need to check for changes in error count.
Maybe we should refactor this test setup in general, but the proposed changes make them brittle. 
We should be able to count how many errors are contained in the new multi error, right?

File: rpm/BUILD.bazel [hunk 2]
```diff
@@ -1,133 +1,8 @@
...
-load("@rules_cc//cc:defs.bzl", "cc_library")
-
-cc_library(
-    name = "cryptsetup",
-    srcs = ["//rpm:cryptsetup-libs/usr/lib64"],
-    hdrs = ["//rpm:cryptsetup-libs/usr/include"],
-    linkstatic = 1,
-    strip_include_prefix = "/rpm/cryptsetup-libs/",
-    visibility = ["//visibility:public"],
-)
```
Comment (datosh) [inline rpm/BUILD.bazel:12]: Is this change related to go upgrade?

**Review by datosh — CHANGES_REQUESTED** [review]

File: rpm/BUILD.bazel (hunk 2 above)
Comment (malt3) [reply rpm/BUILD.bazel:12]: This is a temporary change that I had to make since the RPMs are no longer available at the URLs that are hard coded here. The [bazel PR](https://github.com/edgelesssys/constellation/pull/1186) updates those URLs to be correct again.

Comment (malt3) [reply rpm/BUILD.bazel:12]: The cc_library depends on those RPMs so I just removed it here.

File: cli/internal/cmd/configfetchmeasurements.go [hunk 3]
```diff
@@ -80,8 +81,12 @@ func (cfm *configFetchMeasurementsCmd) configFetchMeasurements(
 
 	cfm.log.Debugf("Loading configuration file from %q", flags.configPath)
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/configfetchmeasurements.go:86]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/create.go [hunk 4]
```diff
@@ -73,8 +73,12 @@ func (c *createCmd) create(cmd *cobra.Command, creator cloudCreator, fileHandler
 
 	c.log.Debugf("Loading configuration file from %q", flags.configPath)
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/create.go:78]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/init.go [hunk 5]
```diff
@@ -104,8 +104,12 @@ func (i *initCmd) initialize(cmd *cobra.Command, newDialer func(validator *cloud
 	i.log.Debugf("Using flags: %+v", flags)
 	i.log.Debugf("Loading configuration file from %q", flags.configPath)
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/init.go:109]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/miniup.go [hunk 6]
```diff
@@ -187,8 +187,12 @@ func (m *miniUpCmd) prepareConfig(cmd *cobra.Command, fileHandler file.Handler)
 	// check for existing config
 	if configPath != "" {
 		conf, err := config.New(fileHandler, configPath, force)
+		var configValidationErr *config.ValidationError
+		if errors.As(err, &configValidationErr) {
+			fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/miniup.go:192]:
```suggestion
			cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/recover.go [hunk 7]
```diff
@@ -80,9 +81,14 @@ func (r *recoverCmd) recover(
 
 	r.log.Debugf("Loading configuration file from %q", flags.configPath)
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/recover.go:86]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/upgradecheck.go [hunk 8]
```diff
@@ -129,8 +129,12 @@ type upgradeCheckCmd struct {
 // upgradePlan plans an upgrade of a Constellation cluster.
 func (u *upgradeCheckCmd) upgradeCheck(cmd *cobra.Command, fileHandler file.Handler, flags upgradeCheckFlags) error {
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/upgradecheck.go:134]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/upgradeexecute.go [hunk 9]
```diff
@@ -70,8 +70,12 @@ func upgradeExecute(cmd *cobra.Command, imageFetcher imageFetcher, upgrader clou
 		return fmt.Errorf("parsing flags: %w", err)
 	}
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/upgradeexecute.go:75]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: cli/internal/cmd/verify.go [hunk 10]
```diff
@@ -76,8 +76,12 @@ func (v *verifyCmd) verify(cmd *cobra.Command, fileHandler file.Handler, verifyC
 
 	v.log.Debugf("Loading configuration file from %q", flags.configPath)
 	conf, err := config.New(fileHandler, flags.configPath, flags.force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline cli/internal/cmd/verify.go:81]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: debugd/internal/cdbg/cmd/deploy.go [hunk 11]
```diff
@@ -68,8 +69,12 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 	streamer := streamer.New(fs)
 	transfer := filetransfer.New(log, streamer, filetransfer.ShowProgress)
 	constellationConfig, err := config.New(fileHandler, configName, force)
+	var configValidationErr *config.ValidationError
+	if errors.As(err, &configValidationErr) {
+		fmt.Fprintln(cmd.ErrOrStderr(), []byte(configValidationErr.LongMessage()))
```
Comment (daniel-weisse) [inline debugd/internal/cdbg/cmd/deploy.go:74]:
```suggestion
		cmd.PrintErrln(configValidationErr.LongMessage())
```

File: internal/config/validation.go [hunk 12]
```diff
@@ -24,22 +23,27 @@ import (
...
+}
+
+// LongMessage prints the errors that occurred during validation in a verbose and user friendly way.
+func (e *ValidationError) LongMessage() string {
+	msg := "Problems validating config file:\n"
+	for _, ve := range e.validationErrMsgs {
+		msg += fmt.Sprintf("\t%s\n", ve)
 	}
-	return nil
+	msg += fmt.Sprintln("Fix the invalid entries or generate a new configuration using `constellation config generate`")
```
Comment (daniel-weisse) [inline internal/config/validation.go:45]:
```suggestion
	msg += "Fix the invalid entries or generate a new configuration using `constellation config generate`\n"
```
Don't think we need `fmt.Sprintln()` here, no?

File: .github/workflows/test-tidy.yml [hunk 13]
```diff
@@ -33,39 +33,18 @@ jobs:
...
-        run: |
-          mods=$(go list -f '{{.Dir}}' -m | xargs)
-          echo "Found mods: $mods"
-          echo "submods=${mods}" >> "$GITHUB_OUTPUT"
+          go-version: "1.20.1"
 
       - name: Go tidy check
         id: tidycheck
-        uses: katexochen/go-tidy-check@45731e0013a976d5d616d79007c7ba52de6ce542 # tag=v1.0.2
+        uses: katexochen/go-tidy-check@94dad4bc99ef68aa96b8042ddf35e877e5b9ace9 # tag=v1.0.2
```
[the remark this answers is not in the document]
Comment (katexochen) [reply .github/workflows/test-tidy.yml:40]: I needed to update the action for Go 1.20. This currently uses a PR commit, but will release and update the hash and version before merge.

File: cli/internal/cloudcmd/rollback.go [hunk 14]
```diff
@@ -53,14 +50,12 @@ type rollbackerQEMU struct {
...
+		retErr = r.client.Destroy(ctx)
 	}
-	err = multierr.Append(err, r.libvirt.Stop(ctx))
-	if err == nil {
-		err = r.client.CleanUpWorkspace()
+	if err := r.libvirt.Stop(ctx); err != nil {
+		return errors.Join(retErr, err)
 	}
-	return err
+	return r.client.CleanUpWorkspace()
```
[the remark this answers is not in the document]
Comment (katexochen) [reply cli/internal/cloudcmd/rollback.go:60]: There is should be no case where we reach that line and retErr isn't nil. There is a bug two lines above. xD

File: internal/config/validation.go (hunk 12 above)
Comment (katexochen) [reply internal/config/validation.go:45]: Hm, maybe this shouldn't end this with a newline anyway? String() like functions normally don't end strings with a newline, and we already print this message with PrintErrln...

File: .github/workflows/test-tidy.yml [hunk 15]
```diff
@@ -33,39 +33,18 @@ jobs:
...
-          (steps.tidycheck.conclusion == 'failure') &&
-          startsWith(github.head_ref, 'renovate/') &&
-          !github.event.pull_request.head.repo.fork
-        shell: bash
-        run: |
-          mods="${{ steps.submods.outputs.submods }}"
-          for mod in $mods; do
-            (cd "$mod"; go mod tidy)
-          done
-
```
[the remark this answers is not in the document]
Comment (katexochen) [reply .github/workflows/test-tidy.yml:68]: I rewrote the action as composite, see https://github.com/katexochen/go-tidy-check/pull/3

File: internal/config/validation.go (hunk 12 above)
Comment (daniel-weisse) [reply internal/config/validation.go]: In any case we probably don't need to wrap the string with `Sprintln`

File: .github/workflows/test-tidy.yml (hunk 13 above)
[the remark this answers is not in the document]
Comment (katexochen) [reply .github/workflows/test-tidy.yml]: Updated to new release.

**Review by daniel-weisse — APPROVED** [review]

File: go.mod [hunk 16]
```diff
@@ -90,8 +90,7 @@ require (
 	github.com/spf13/afero v1.9.3
 	github.com/spf13/cobra v1.6.1
 	github.com/stretchr/testify v1.8.1
-	go.uber.org/goleak v1.2.1
-	go.uber.org/multierr v1.9.0
+	go.uber.org/goleak v1.2.0
```
Comment (datosh) [inline go.mod:93]: Was this an intended downgrade?

Comment (katexochen) [reply go.mod:93]: nope.

**Review by datosh — APPROVED** [review]
